test/assignment.coffee

```diff
@@ -131,6 +131,12 @@ test "more compound assignment", ->
   val ?= true
   eq c, val
 
+test "#1192: assignment starting with object literals", ->
+  doesNotThrow (-> CoffeeScript.run "{}.p = 0")
+  doesNotThrow (-> CoffeeScript.run "{}.p++")
+  doesNotThrow (-> CoffeeScript.run "{}[0] = 1")
```

Comment:
Added. Though I removed the first `[0]` assuming that’s what you meant. Correct me if I’m wrong.